### PR TITLE
Add go-task

### PR DESCRIPTION
**Pull request by AntoinePrv**
Follow up to #20518

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by AntoinePrv** [comment]
@dbast , as you had a look on the first version of this, would you mind having a second look?

**Comment by AntoinePrv** [comment]
@conda-forge/staged-recipes could you please review this Go recipe.

File: recipes/go-task/meta.yaml
Comment (xhochy) [inline recipes/go-task/meta.yaml:5]: What's the reason to not use purely `src`?

File: recipes/go-task/meta.yaml
Comment (xhochy) [inline recipes/go-task/meta.yaml:25]: Please use either go-nocgo or go-cgo here.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/go-task/meta.yaml
Comment (AntoinePrv) [reply recipes/go-task/meta.yaml:25]: I added `go-nocgo` but kept the `{{ compiler('go') }}` as I was not working otherwise.

File: recipes/go-task/meta.yaml
Comment (AntoinePrv) [reply recipes/go-task/meta.yaml:5]: I am no go developer but my understanding is that go has strict directory requirements and expects a full namespace like `src/github.com/go-task`.
I took inspiration from [the kubo feedstock](https://github.com/conda-forge/kubo-feedstock/blob/d070776c6e917bf1c699a39f5aeb2f7457c37f34/recipe/meta.yaml#L11) as suggested by @dbast.

**Comment by AntoinePrv** [comment]
@xhochy would you have more time to make a second review?

**Comment by wolfv** [comment]
Looks good to me. I checked a couple other recipes and `compiler('go')` seems to resovle to `go-nocgo` these days (based on the conda-forge pinnings of `go_compiler`. So this looks fine! :)

**Pull request merged**
